internal/schema/network_peer.go

```diff
@@ -229,6 +260,20 @@ func morphToProviderConfig(networkPeer *network_peer_api.GetNetworkPeeringRecord
 	} else if err != nil {
 		return ProviderConfig{}, fmt.Errorf("%s: %w", errors.ErrReadingGCPConfig, err)
 	}
+	azure, err := networkPeer.AsAZURE()
+	if err == nil && azure.AzureConfigData.AzureTenantId != "" {
+		newProviderConfig.AzureConfig = &AzureConfig{
+			ProviderId:     types.StringValue(azure.ProviderId),
+			Cidr:           types.StringValue(azure.AzureConfigData.Cidr),
+			SubscriptionId: types.StringValue(azure.AzureConfigData.SubscriptionId),
+			ResourceGroup:  types.StringValue(azure.AzureConfigData.ResourceGroup),
+			VnetId:         types.StringValue(azure.AzureConfigData.VnetId),
+			AzureTenantId:  types.StringValue(azure.AzureConfigData.AzureTenantId),
+		}
+		return newProviderConfig, nil
+	} else if err != nil {
+		return ProviderConfig{}, fmt.Errorf("%s: %w", errors.ErrReadingAzureConfig, err)
+	}
```

Comment:
If err == nil and azure.AzureConfigData.AzureTenantId is an empty string, should we return an error in this case?, seems like this case is missing.

We can also use a switch case statement instead of an if-else statement in this case.